### PR TITLE
Added doc for Val/Eval and lm_eval integration

### PR DESCRIPTION
**Pull request by wesleytruong**
This pr adds documentation for how to get started with
- In training validation in TorchTitan
- Third party evaluation with `lm_eval`

### CODE REVIEW

File: torchtitan/models/llama3/train_configs/llama3_8b.toml [hunk 1]
```diff
@@ -60,3 +60,8 @@ selective_ac_option = "op"  # "int" = ac every positive int layer or 'op', ac ba
 enable_fsdp_float8_all_gather = false
 precompute_float8_dynamic_scale_for_fsdp = false
 filter_fqns = ["output"]
+
+[validation]
+enabled = true
+freq = 5
```
Comment (tianyu-l) [inline torchtitan/models/llama3/train_configs/llama3_8b.toml]: this is too frequent, maybe set to 100?

File: torchtitan/models/llama3/train_configs/llama3_8b.toml [hunk 2]
```diff
@@ -60,3 +60,8 @@ selective_ac_option = "op"  # "int" = ac every positive int layer or 'op', ac ba
...
+steps = 10
```
Comment (tianyu-l) [inline torchtitan/models/llama3/train_configs/llama3_8b.toml]: I think we can afford -1

File: docs/evaluation.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
+# Validation and Evaluation in HF
```
Comment (tianyu-l) [inline docs/evaluation.md]:
```suggestion
# Validation and Evaluation
```

File: docs/evaluation.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+`torchtitan` provides direct and indirect support for validation to support user's training goals. Direct support is provided by the `Validator` class which interacts directly with the training loop, and indirect support is provided through [HuggingFace checkpoint conversion](https://github.com/pytorch/torchtitan/blob/main/docs/checkpoint.md#huggingface) for users who want to asynchronously do evaluation using external tools such as ELeutherAI's `lm_eval`.
```
Comment (tianyu-l) [inline docs/evaluation.md]:
```suggestion
`torchtitan` provides direct and indirect support for validation to support user's training goals. Direct support is provided by the `Validator` class which interacts directly with the training loop, and indirect support is provided through [HuggingFace checkpoint conversion](https://github.com/pytorch/torchtitan/blob/main/docs/checkpoint.md#huggingface) for users who want to do evaluation using external tools such as ELeutherAI's `lm_eval`.
```

File: docs/evaluation.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+## Third-Party Evaluation
+With `./scripts/checkpoint_conversion/convert_to_hf`, `torchtitan` offers support for converting checkpoints from DCP to safetensors format. Using this script, users can perform efficient evaluation separate from their training using external libraries that support HuggingFace e.g. `lm_eval` with `vllm` backend.
+
+### Example usage of `lm_eval` with `vllm`:
+To use this specific setup make sure to include a HuggingFace `config.json` file which is not provided by conversion script or `last_save_in_hf` option. The HF config file can be downloaded by running `python ./scripts/download_hf_assets.py --repo_id meta-llama/Llama-3.1-8B --assets config`.
+
+Note that pip installing `lm-eval` may result in breaking `torchtitan` dev environment so we recommend creating a separate env.
+```bash
+pip install "lm-eval[vllm]"
+lm_eval --model vllm \
```
Comment (tianyu-l) [inline docs/evaluation.md:27]: could you include example eval results here?

File: docs/evaluation.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+For users who want to perform validation directly during the training loop, we provide the `Validator` class which can be conveniently overloaded through the `TrainSpec` or configured in `JobConfig`. The validator class has access to and reuses many of the trainer's functions such as its parallelization, including pipelining.
+
+Below is an example validation config:
+
+```toml
+[validation]
+enabled = true
+dataset = "c4_validation"
+freq = 500
+steps = -1 #consume the entire validation set
```
Comment (tianyu-l) [inline docs/evaluation.md]:
```suggestion
steps = -1  # consumes the entire validation set
```

File: docs/evaluation.md [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+```toml
+[validation]
+enabled = true
+dataset = "c4_validation"
+freq = 500
+steps = -1 # consumes the entire validation set
+```
+
+## Third-Party Evaluation
+With `./scripts/checkpoint_conversion/convert_to_hf`, `torchtitan` offers support for converting checkpoints from DCP to safetensors format. Using this script, users can perform efficient evaluation separate from their training using external libraries that support HuggingFace e.g. `lm_eval` with `vllm` backend.
```
Comment (wwwjn) [inline docs/evaluation.md]:
```suggestion
With `./scripts/checkpoint_conversion/convert_to_hf.py`, `torchtitan` offers support for converting checkpoints from DCP to safetensors format. Using this script, users can perform efficient evaluation separate from their training using external libraries that support HuggingFace e.g. `lm_eval` with `vllm` backend.
```

File: torchtitan/models/llama3/train_configs/llama3_8b.toml [hunk 8]
```diff
@@ -60,3 +60,8 @@ selective_ac_option = "op"  # "int" = ac every positive int layer or 'op', ac ba
 enable_fsdp_float8_all_gather = false
 precompute_float8_dynamic_scale_for_fsdp = false
 filter_fqns = ["output"]
+
+[validation]
+enabled = false
```
Comment (wwwjn) [inline torchtitan/models/llama3/train_configs/llama3_8b.toml:65]: Why we don't add `validation.dataset` here?
